From 2f7507e1be0324cffcab56ad45bc7984078b9859 Mon Sep 17 00:00:00 2001
From: skomakul <quic_skomakul@quicinc.com>
Date: Mon, 20 Jan 2025 14:02:45 +0530
Subject: [PATCH] gstd: update python pip command.

Upstream-Status: Inappropriate

Signed-off-by: Shyam Komakula  <quic_skomakul@quicinc.com>
---
 libgstc/python/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgstc/python/meson.build b/libgstc/python/meson.build
index 057114b..f0c4556 100644
--- a/libgstc/python/meson.build
+++ b/libgstc/python/meson.build
@@ -1 +1 @@
-run_command('pip3', 'install', '.')
+run_command('python3', '-m', 'pip', 'install', '')
